rs/index/src/segment/mutable_segment.rs

```diff
@@ -81,6 +105,14 @@ impl MutableSegment {
         Ok(())
     }
 
+    pub fn start_being_flushed(&mut self) {
+        self.use_invalidated_id_map = true;
+    }
```

Comment:
wait - this function is only for tests, why are you mixing it with actual logic?